### PR TITLE
[SYNTH11931] Create README for the integration

### PR DESCRIPTION
**Pull request by teodor2312**
Fixing the README file for this integration to show relevant information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,27 @@
-# datadog-mobile-app-run-tests
+# Datadog Continuous Testing for Bitrise
```
Comment (Drarig29) [inline README.md:1]: Nice! It's the full name indeed ([docs](https://docs.datadoghq.com/continuous_testing/)), and we use that kind of title for Azure DevOps: https://github.com/DataDog/datadog-ci-azure-devops

Should we align that title for all CI integrations? cc. @jakepruitt

File: README.md [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
 
-placeholder description
+<!-- TODO add link to marketplace after we publish the step -->
```
Comment (Drarig29) [inline README.md:3]: Can you add this to the "Definition of done" of the corresponding card?

File: README.md [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+<!-- [![Visual Studio Marketplace Version]()][A-1]  -->
+<!-- [![Build Status](https://dev.azure.com/datadog-ci/Datadog%20CI%20Azure%20DevOps%20Extension/_apis/build/status%2FDevelopment?branchName=main)](https://dev.azure.com/datadog-ci/Datadog%20CI%20Azure%20DevOps%20Extension/_build/latest?definitionId=4&branchName=main) -->
```
Comment (Drarig29) [inline README.md:5]: Can you change that icon to show the Bitrise CI status?

https://devcenter.bitrise.io/en/apps/embedding-an-app-status-badge.html

File: README.md [hunk 4]
```diff
@@ -31,63 +42,153 @@ envs:
...
+[C-6]: https://docs.datadoghq.com/account_management/api-app-keys/
+[C-7]: https://docs.datadoghq.com/developers/guide/what-best-practices-are-recommended-for-naming-metrics-and-tags/#rules-and-best-practices-for-naming-tags
+
+<!-- Integration specific links -->
+[D-1]: https://devcenter.bitrise.io/en/steps-and-workflows/introduction-to-steps/adding-steps-to-a-workflow.html
+[D-2]: https://devcenter.bitrise.io/en/builds/secrets.html#setting-a-secret
+[D-3]: https://devcenter.bitrise.io/en/steps-and-workflows/introduction-to-steps/step-inputs.html
+
+<!-- Other -->
+[E-1]: https://www.datadoghq.com/blog/best-practices-datadog-continuous-testing/
```
Comment (Drarig29) [inline README.md:194]: I didn't know that was possible! We have to ensure it's supported by our Datadog docs, since this README page will at some point be single-sourced (i.e. used as the source code for the Datadog docs as well).

[**Example:**](https://docs.datadoghq.com/continuous_testing/cicd_integrations/github_actions/)

image

File: README.md [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
-Can be run directly with the [bitrise CLI](https://github.com/bitrise-io/bitrise),
-just `git clone` this repository, `cd` into it's folder in your Terminal/Command Line
-and call `bitrise run test`.
+## Setup
 
-*Check the `bitrise.yml` file for required inputs which have to be
-added to your `.bitrise.secrets.yml` file!*
+To get started:
+
+1. Add this step to your workflow. You can see Bitrise's documentation on how to do that [here][D-1]. You can also configure it locally by referencing this step in your `bitrise.yml`
```
Comment (Drarig29) [inline README.md:16]:
```suggestion
1. Add this step to your workflow. You can see Bitrise's documentation on how to do that [here][D-1]. You can also configure it locally by referencing this step in your `bitrise.yml`.
```

File: README.md [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
+2. Add you API and App Keys to your secrets in Bitrise. Documentation on how to do that can be found [here][D-2]
```
Comment (Drarig29) [inline README.md:17]:
```suggestion
2. Add you API and App Keys to your secrets in Bitrise. Documentation on how to do that can be found [here][D-2].
```

File: README.md [hunk 7]
```diff
@@ -1,16 +1,27 @@
...
+3. Configure your step inputs ([docs][D-3]). You can also configure them in your `bitrise.yml`. The only required inputs are the 2 secrets you configured earlier. The rest of the possible inputs will be described later on.
+
+## How to use this Step locally
+
+Can be run directly with the [bitrise CLI][B-3], just `git clone` this repository, `cd` into it's folder in your Terminal/Command Line and call `bitrise run test`.
```
Comment (Drarig29) [inline README.md:22]: Looks like `git clone` and `cd` are already described in the step-by-step process below.

File: README.md [hunk 8]
```diff
@@ -31,63 +42,153 @@ envs:
...
-2. Copy the [step template](https://github.com/bitrise-steplib/step-template) files into your repository
-3. Fill the `step.sh` with your functionality
-4. Wire out your inputs to `step.yml` (`inputs` section)
-5. Fill out the other parts of the `step.yml` too
-6. Provide test values for the inputs in the `bitrise.yml`
-7. Run your step with `bitrise run test` - if it works, you're ready
+### Example using public IDs
 
-__For Step development guidelines & best practices__ check this documentation: [https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md).
+<!-- TODO: change git urls to step references after we publish it -->
```
Comment (Drarig29) [inline README.md:49]: Also add that to the corresponding card (the definition of done can just say something like "fix all TODOs in READMEs")

File: README.md [hunk 9]
```diff
@@ -31,63 +42,153 @@ envs:
...
+```yml
+- git::https://github.com/DataDog/synthetics-test-automation-bitrise-step-run-tests.git:
+   inputs:
+   - api_key: $DATADOG_API_KEY
+   - app_key: $DATADOG_APP_KEY
+   - public_ids: '012-345-678, 9ab-cde-fff'
```
Comment (Drarig29) [inline README.md:55]: I'm not a fan of the "only-numbers" one... can you use the same example as the other integrations?

```suggestion
   - public_ids: 'abc-d3f-ghi, jkl-mn0-pqr'
```

File: README.md [hunk 10]
```diff
@@ -31,63 +42,153 @@ envs:
...
+- git::https://github.com/DataDog/synthetics-test-automation-bitrise-step-run-tests.git:
    inputs:
-   - my_input_1: "my value 1"
-   - my_input_2: "my value 2"
+   - api_key: $DATADOG_API_KEY
+   - app_key: $DATADOG_APP_KEY
+   - test_search_query: 'tag:e2e-tests'
+   - variables: |
+      START_URL=https://staging.website.com
+      PASSWORD=$(StagingPassword)
```
Comment (Drarig29) [inline README.md:92]: I doubt that `$(StagingPassword)` is the actual syntax to refer to secrets in Bitrise 🤔 

Also the `StagingPassword` camel case is really "Microsoft-y" 😅 

Something like this would probably work:

```suggestion
      PASSWORD=$STAGING_PASSWORD
```

File: README.md [hunk 11]
```diff
@@ -31,63 +42,153 @@ envs:
...
+   - app_key: $DATADOG_APP_KEY
+   - config_path: './synthetics-config.json'
+```
+
+For an example configuration file, see this [`global.config.json` file][B-1].
 
+### Example including all possible configurations
 
-## Share your own Step
+There shouldn't be any cases where you would need to use all of the configurations at the same time as for example `public_ids` and `test_search_query` are mutually exclusive. However having an example for the value of each is useful.
```
Comment (Drarig29) [inline README.md:111]: No need to be too wordy about that, and I believe that in a perfect world the `public_ids` and `test_search_query` shouldn't be mutually exclusive. For now this is an implementation detail, which can change in datadog-ci 😉 

```suggestion
For reference, here is how a full configuration could look like:
```

**Comment by Drarig29** [comment]
@teodor2312 Once out of draft, can you add the @\DataDog/documentation team as reviewers?

You can also add them in the `CODEOWNERS` file ([example](https://github.com/DataDog/synthetics-ci-github-action/blob/main/CODEOWNERS)).

File: README.md (hunk 1 above)
Comment (teodor2312) [reply README.md:1]: Yup i read the other 3 integration READMEs and Frankenstein-ed this one from the best parts of them in my opinion :D

File: README.md (hunk 9 above)
Comment (teodor2312) [reply README.md:55]: Will do, i released i changed them based on a false premise 🤦

**Comment by teodor2312** [comment]
@DataDog/documentation can you also please have a look at this PR :)

**Review by Drarig29 — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (jhgilbert) [inline README.md]:
```suggestion
1. Add this step to your workflow. You can see Bitrise's documentation on how to do that [here][D-1]. You can also configure it locally by referencing this step in your `bitrise.yml`.
```

For accessibility reasons, the text of a link should describe the content of the link. For example, you could rewrite the sentence above to "See Bitrise's documentation on how to do that," with "Bitrise's documentation" as the link.

File: README.md [hunk 12]
```diff
@@ -1,16 +1,27 @@
...
-and call `bitrise run test`.
+## Setup
 
-*Check the `bitrise.yml` file for required inputs which have to be
-added to your `.bitrise.secrets.yml` file!*
+To get started:
+
+1. Add this step to your workflow. You can see Bitrise's documentation on how to do that [here][D-1]. You can also configure it locally by referencing this step in your `bitrise.yml`
+2. Add you API and App Keys to your secrets in Bitrise. Documentation on how to do that can be found [here][D-2]
+3. Configure your step inputs ([docs][D-3]). You can also configure them in your `bitrise.yml`. The only required inputs are the 2 secrets you configured earlier. The rest of the possible inputs will be described later on.
```
Comment (jhgilbert) [inline README.md:18]:
```suggestion
3. Configure your step inputs ([docs][D-3]). You can also configure them in your `bitrise.yml`. The only required inputs are the 2 secrets you configured earlier. The rest of the possible inputs are described in a later section.
```

Unfortunately, while references like [D-3] are supported by Markdown, using letters in link references can cause problems in the documentation translation system. If you would still like to have clear sections in your links, you could use 1000, 2000, 3000, etc., with 1003 being the third link in the 1000 series.

File: README.md (hunk 7 above)
Comment (jhgilbert) [inline README.md]:
```suggestion
Can be run directly with the [Bitrise CLI][B-3], just `git clone` this repository, `cd` into it's folder in your Terminal/Command Line and call `bitrise run test`.
```
Or `bitrise`

File: README.md (hunk 11 above)
Comment (jhgilbert) [inline README.md]:
```suggestion
You won't need to use all of the configurations at the same time (for example, `public_ids` and `test_search_query` are mutually exclusive). The example below is not intended as a valid configuration, it just provides examples of all values for reference.
```

File: README.md [hunk 13]
```diff
@@ -31,63 +42,153 @@ envs:
...
+| `appKey`                           | _required_  | Your Datadog application key. This key is created by your [Datadog organization][C-6] and will be accessed as an environment variable.                                                                                |
+| `configPath`                       | _optional_  | The global JSON configuration is used when launching tests. See the [example configuration][C-3] for more details.                                                                                                    |
+| `deviceIds`                        | _optional_  | Override the mobile device(s) to run your mobile test.                                                                                                                                                                |
+| `failOnCriticalErrors`             | _optional_  | A boolean flag that fails the CI job if no tests were triggered, or results could not be fetched from Datadog. The default is set to `false`                                                                          |
```
Comment (jhgilbert) [inline README.md:145]:
```suggestion
| `failOnCriticalErrors`             | _optional_  | A boolean flag that fails the CI job if no tests were triggered, or results could not be fetched from Datadog. The default is set to `false`.                                                                          |
```

File: README.md (hunk 6 above)
Comment (jhgilbert) [inline README.md]:
```suggestion
2. Add your API and App Keys to your secrets in Bitrise. Documentation on how to do that can be found [here][D-2].
```

File: README.md (hunk 7 above)
Comment (jhgilbert) [inline README.md]:
```suggestion
This step can be run directly with the [bitrise CLI][B-3]. Just `git clone` this repository, `cd` into the repository folder, and call `bitrise run test`.
```

File: README.md [hunk 14]
```diff
@@ -21,7 +32,7 @@ Step by step:
    (the `.bitrise.secrets.yml` is a git ignored file, you can store your secrets in it)
 6. Check the `bitrise.yml` file for any secret you should set in `.bitrise.secrets.yml`
   * Best practice is to mark these options with something like `# define these in your .bitrise.secrets.yml`, in the `app:envs` section.
-7. Once you have all the required secret parameters in your `.bitrise.secrets.yml` you can just run this step with the [bitrise CLI](https://github.com/bitrise-io/bitrise): `bitrise run test`
+7. Once you have all the required secret parameters in your `.bitrise.secrets.yml` you can just run this step with the [bitrise CLI][B-3]: `bitrise run test`
```
Comment (jhgilbert) [inline README.md:32]:
```suggestion
7. Once you have the required secret parameters in your `.bitrise.secrets.yml`, run this step with the [bitrise CLI][B-3]: `bitrise run test`.
```

File: README.md [hunk 15]
```diff
@@ -31,63 +42,153 @@ envs:
...
-7. Once you're done just commit your changes & create a Pull Request
+```yml
+- git::https://github.com/DataDog/synthetics-test-automation-bitrise-step-run-tests.git:
+   inputs:
+   - api_key: $DATADOG_API_KEY
+   - app_key: $DATADOG_APP_KEY
+   - config_path: './synthetics-config.json'
+```
+
+For an example configuration file, see this [`global.config.json` file][B-1].
```
Comment (jhgilbert) [inline README.md:104]:
```suggestion
For an example configuration file, see the [`global.config.json` file][B-1].
```

File: README.md [hunk 16]
```diff
@@ -31,63 +42,153 @@ envs:
...
+| `configPath`                       | _optional_  | The global JSON configuration is used when launching tests. See the [example configuration][C-3] for more details.                                                                                                    |
+| `deviceIds`                        | _optional_  | Override the mobile device(s) to run your mobile test.                                                                                                                                                                |
+| `failOnCriticalErrors`             | _optional_  | A boolean flag that fails the CI job if no tests were triggered, or results could not be fetched from Datadog. The default is set to `false`                                                                          |
+| `failOnMissingTests`               | _optional_  | Fail the CI job if at least one specified test with a public ID (using `publicIds` or listed in a [test file][C-2] is missing in a run (for example, if it has been deleted programmatically or on the Datadog site). |
```
Comment (jhgilbert) [inline README.md:146]:
```suggestion
| `failOnMissingTests`               | _optional_  | Fail the CI job if at least one specified test with a public ID (using `publicIds` or listed in a [test file][C-2]) is missing in a run (for example, if it has been deleted programmatically or on the Datadog site). |
```

File: README.md [hunk 17]
```diff
@@ -31,63 +42,153 @@ envs:
...
+| `locations`                        | _optional_  | String of locations separated by semicolons to override the locations where your tests run.                                                                                                                           |
+| `mobileApplicationVersion`         | _optional_  | Override the default mobile application version for a Synthetic mobile application test. The version must be uploaded and available within Datadog.                                                                   |
+| `mobileApplicationVersionFilePath` | _optional_  | Override the application version for Synthetic mobile application tests.                                                                                                                                              |
+| `pollingTimeout`                   | _optional_  | The duration (in milliseconds) after which the batch is deemed as failed because of a time out. The default is 30 minutes.                                                                                            |
```
Comment (jhgilbert) [inline README.md:153]:
```suggestion
| `pollingTimeout`                   | _optional_  | The duration (in milliseconds) after which the batch is deemed as failed because of a timeout. The default is 30 minutes.                                                                                            |
```

File: README.md [hunk 18]
```diff
@@ -31,63 +42,153 @@ envs:
...
+| `publicIds`                        | _optional_  | String of public IDs separated by commas for Synthetic tests you want to trigger.                                                                                                                                     |
+| `site`                             | _optional_  | The Datadog site to send data to. If the `DD_SITE` environment variable is set, it takes precedence.                                                                                                                  |
+| `subdomain`                        | _optional_  | The name of the custom subdomain set to access your Datadog application. If the URL used to access Datadog is `myorg.datadoghq.com`, the `subdomain` value needs to be set to `myorg`.                                |
+| `testSearchQuery`                  | _optional_  | Trigger tests corresponding to a [search][C-5]  query. This can be useful if you are tagging your test configurations. See [best practices][C-7]  for more information on tagging.                                    |
```
Comment (jhgilbert) [inline README.md:157]:
```suggestion
| `testSearchQuery`                  | _optional_  | Trigger tests corresponding to a [search][C-5] query. This can be useful if you are tagging your test configurations. See [best practices][C-7]  for more information on tagging.                                    |
```

**Review by jhgilbert — COMMENTED** [review]
I left some initial comments, I'll loop back for another pass. Thank you!

**Comment by jhgilbert** [comment]
@teodor2312 A lot of the feedback I gave here also applies to [the other PR](https://github.com/DataDog/synthetics-test-automation-bitrise-step-upload-application/pull/6/files) -- I'll wait to review that one, just so you aren't getting the same comments twice. Thank you!

**Comment by teodor2312** [comment]
Hi @jhgilbert I addressed your feedback 👍 thanks for the comments and the suggestion for how to make the references better :) 

I think you were using an outdated version of the PR somehow where you were commenting though 🤔 maybe you hadn't refreshed the page or were only looking at the 1st commit

File: README.md [hunk 19]
```diff
@@ -1,27 +1,35 @@
...
-placeholder description
+<!-- TODO add link to marketplace after we publish the step -->
+<!-- [![Visual Studio Marketplace Version]()][1001]  -->
+[![Build Status](https://app.bitrise.io/app/7846c17b-8a13004fc7-aced-5f3b0b2ec6c4/status.svg?token=480MdFpG78E6kZASg5w1dw&branch=main)](https://app.bitrise.io/app/7846c17b-8a13004fc7-aced-5f3b0b2ec6c4)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
+# Overview
 
-## How to use this Step
+With the `synthetics-test-automation-bitrise-step-run-tests` step you can run Synthetics tests during your Bitrise CI and ensure all your teams using Bitrise can benefit from Synthetic tests at every stage of the software lifecycle. This step uses the [Datadog CI Synthetics command][2002].
```
Comment (jhgilbert) [inline README.md:10]:
```suggestion
With the `synthetics-test-automation-bitrise-step-run-tests` step, you can run Synthetics tests during your Bitrise CI and ensure all your teams using Bitrise can benefit from Synthetic tests at every stage of the software lifecycle. This step uses the [Datadog CI Synthetics command][2002].
```

File: README.md [hunk 20]
```diff
@@ -1,27 +1,35 @@
...
-and call `bitrise run test`.
+## Setup
 
-*Check the `bitrise.yml` file for required inputs which have to be
-added to your `.bitrise.secrets.yml` file!*
+To get started:
 
-Step by step:
+1. Add this step to your workflow. You can see [Bitrise's documentation][4001] on how to do that. You can also configure it locally by referencing this step in your `bitrise.yml`.
+2. Add your API and App Keys to your secrets in Bitrise. Documentation on how to do that can be found here - [Setting a Secret][4002].
```
Comment (jhgilbert) [inline README.md:17]:
```suggestion
2. Add your API and App keys to your secrets in Bitrise. Documentation on how to do that can be found here: [Setting a Secret][4002].
```

File: README.md [hunk 21]
```diff
@@ -1,27 +1,35 @@
...
+3. Configure your step inputs ([Step Inputs][4003]). You can also configure them in your `bitrise.yml`. The only required inputs are the 2 secrets you configured earlier. The rest of the possible inputs are described in a later section.
```
Comment (jhgilbert) [inline README.md:18]:
```suggestion
3. Configure your step inputs (see the [Step Inputs documentation][4003]). You can also configure them in your `bitrise.yml`. The only required inputs are the 2 secrets you configured earlier. The rest of the possible inputs are described in a later section.
```

File: README.md [hunk 22]
```diff
@@ -1,27 +1,35 @@
...
+
+## How to use this Step locally
```
Comment (jhgilbert) [inline README.md:20]:
```suggestion
## How to use this step locally
```
It seems that "step" is lowercased most of the time, so I'm lowercasing it here for consistency, but you could also capitalize it throughout if you prefer that.

File: README.md [hunk 23]
```diff
@@ -1,27 +1,35 @@
...
+
+This Step can be run directly with the [Bitrise CLI][2003].
```
Comment (jhgilbert) [inline README.md:22]:
```suggestion
This step can be run directly with the [Bitrise CLI][2003].
```

**Review by jhgilbert — APPROVED** [review]
Sorry about the outdated review earlier, not sure what happened there! I reviewed the current version and am approving with some minor suggestions. Thanks!

File: README.md [hunk 24]
```diff
@@ -31,63 +39,153 @@ envs:
...
-  * (relative) path format: instead of `- original-step-id:` use `- path::./relative/path/of/script/on/your/Mac:`
-  * direct git URL format: instead of `- original-step-id:` use `- git::https://github.com/user/step.git@branch:`
-  * You can find more example of alternative step referencing at: https://github.com/bitrise-io/bitrise/blob/master/_examples/tutorials/steps-and-workflows/bitrise.yml
-7. Once you're done just commit your changes & create a Pull Request
+For an example configuration file, see the [`global.config.json` file][2001].
 
+### Example including all possible configurations
 
-## Share your own Step
+For reference here's how a full configuration could look like:
```
Comment (jhgilbert) [inline README.md:108]:
```suggestion
For reference here's how a full configuration could look:
```

File: README.md (hunk 22 above)
Comment (teodor2312) [reply README.md:20]: I think this one i'll leave capitalised as `Step` is the terminology that Bitrise uses for this kind of integration
